cmd/tailscale/cli/serve.go

```diff
@@ -58,13 +117,44 @@ func newServeCommand(e *serveEnv) *ffcli.Command {
 // It also contains the flags, as registered with newServeCommand.
 type serveEnv struct {
 	// flags
+	servePort    uint // Port to serve on. Defaults to 443.
 	terminateTLS bool
+	remove       bool // remove a serve config
+	json         bool // output JSON (status only for now)
 
 	// optional stuff for tests:
-	testFlagOut        io.Writer
-	testGetServeConfig func(context.Context) (*ipn.ServeConfig, error)
-	testSetServeConfig func(context.Context, *ipn.ServeConfig) error
-	testStdout         io.Writer
+	testFlagOut              io.Writer
+	testGetServeConfig       func(context.Context) (*ipn.ServeConfig, error)
+	testSetServeConfig       func(context.Context, *ipn.ServeConfig) error
+	testGetLocalClientStatus func(context.Context) (*ipnstate.Status, error)
```

Comment:
I think it might be better to have a fake localClient instead of these. 
Maybe adding a `type localServeClient interface{}` with the required functions to facilitate.

Can happen in a follow up.

Reply:
#6372 created